crates/sequencer/src/mempool.rs

```diff
@@ -24,16 +25,56 @@ type Transaction<C> = <CitreaMempoolImpl<C> as TransactionPool>::Transaction;
 pub(crate) struct CitreaMempool<C: sov_modules_api::Context>(CitreaMempoolImpl<C>);
 
 impl<C: sov_modules_api::Context> CitreaMempool<C> {
-    pub(crate) fn new(client: DbProvider<C>) -> Self {
+    pub(crate) fn new(client: DbProvider<C>, mempool_conf: SequencerMempoolConfig) -> Self {
         let blob_store = NoopBlobStore::default();
-        let genesis_hash = client.genesis_block().unwrap().unwrap().header.hash;
+        let genesis_block = client.genesis_block().unwrap().unwrap();
         let evm_config = client.cfg();
-        let chain_spec = ChainSpec {
-            chain: Chain::from_id(evm_config.chain_id),
-            genesis_hash,
-            base_fee_params: BaseFeeParamsKind::Constant(evm_config.base_fee_params),
-            ..Default::default()
+
+        let chain_spec = ChainSpecBuilder::default()
+            .chain(Chain::from_id(evm_config.chain_id))
+            .shanghai_activated()
+            .genesis(
+                Genesis::default()
+                    .with_nonce(genesis_block.header.nonce.unwrap().into())
+                    .with_timestamp(genesis_block.header.timestamp.saturating_to())
+                    .with_extra_data(genesis_block.header.extra_data)
+                    .with_gas_limit(genesis_block.header.gas_limit.saturating_to())
+                    .with_difficulty(genesis_block.header.difficulty)
+                    .with_mix_hash(genesis_block.header.mix_hash.unwrap())
+                    .with_coinbase(genesis_block.header.miner)
+                    .with_base_fee(Some(
+                        genesis_block
+                            .header
+                            .base_fee_per_gas
+                            .unwrap()
+                            .saturating_to(),
+                    )),
+            )
+            .build();
+
+        // Default 10x'ed from standard limits
+        let pool_config = Default::default();
+        let pool_config = PoolConfig {
+            pending_limit: SubPoolLimit {
+                max_txs: mempool_conf.pending_tx_limit as usize,
+                max_size: TXPOOL_SUBPOOL_MAX_SIZE_MB_DEFAULT * 10 * 1024 * 1024,
```

Comment:
```
  88   │ impl Default for SubPoolLimit {
  89   │     fn default() -> Self {
  90   │         // either 10k transactions or 20MB
  91   │         Self {
  92   │             max_txs: TXPOOL_SUBPOOL_MAX_TXS_DEFAULT,
  93   │             max_size: TXPOOL_SUBPOOL_MAX_SIZE_MB_DEFAULT * 1024 * 1024,
  94   │         }
  95   │     }
  96   │ }
```

So better use syntax:

```
SubPoolLimit {
    max_txs: mempool_conf.pending_tx_limit as usize,
    ..Default::default()
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=97a6d8b64de9512439aa424852fc11bf

Reply:
I see your point. That was my first attempt too but at the same time the Default seems to be wrong - it works as if it's KB rather than MB. 

I pointed this to Esad and tested on my machine, and will open a PR to Reth (if I am not completely mistaken, will check again)